core/trino-main/src/main/java/io/trino/operator/scalar/SequenceFunction.java

```diff
@@ -120,6 +121,34 @@ private static Block fixedWidthSequence(long start, long stop, long step, FixedW
         return blockBuilder.build();
     }
 
+    private static int getLength(long start, long stop, long step)
+    {
+        if ((step > 0 && (start >= 0 || stop <= 0))
+                || (step < 0 && (start <= 0 || stop >= 0))) {
+            long length = (stop - start) / step + 1L;
+            checkMaxEntry(length);
+            return toIntExact(length);
+        }
+
+        checkMaxEntry(start / step);
+        checkMaxEntry(stop / step);
+        int startLength = abs(toIntExact(start / step));
+        int stopLength = abs(toIntExact(stop / step));
+
+        // 1. step > 0: -step ... startRemain ... 0 ... stopRemain ... step
+        // 2. step < 0: step ... stopRemain ... 0 ... startRemain ... -step
+        long startRemain = start % step;
+        long stopRemain = stop % step;
+        int remainLength = 1;
+        if (step > 0) {
+            remainLength += startRemain + step <= stopRemain ? 1 : 0;
+        }
+        else {
+            remainLength += startRemain + step >= stopRemain ? 1 : 0;
+        }
+        return startLength + stopLength + remainLength;
```

Comment:
Do `checkMaxEntry(startLength + stopLength + remainLength)` before you return. Add a test case where this check fails.

Reply:
I think this one should be covered by https://github.com/trinodb/trino/pull/16867/files#diff-c97e3bbca82a66977fec2ad333e149a49f1a10f35bbee472693cf6f6af7a3a42L4312 already